test/plugin/base_test.rb

```diff
@@ -1275,10 +1296,17 @@ def http_request_message
     _undefined
   end
 
-  # A wrapper around the constant HTTP_REQUEST_MESSAGE_WITHOUT_REFERER, so the
-  # definition can be skipped in the shared module and defined in the test
-  # classes later.
-  def http_request_message_without_referer
+  # A wrapper around the constant HTTP_REQUEST_MESSAGE_EXPECTED_WITHOUT_REFERER,
+  # so the definition can be skipped in the shared module and defined in the
+  # test classes later.
+  def http_request_message_expected_without_referer
+    _undefined
+  end
+
+  # A wrapper around the constant HTTP_REQUEST_MESSAGE_EXPECTED_STRUCT_PAYLOAD,
+  # so the definition can be skipped in the shared module and defined in the
+  # test classes later.
+  def http_request_message_expected_struct_payload
```

Comment:
Instead of this, why don't we define `assert_null_struct` as a virtual method? But see the other comment about whether this is the right behavior in the first place.